main/src/com/google/refine/importers/tree/XmlImportUtilities.java

```diff
@@ -356,23 +357,29 @@ static protected void findRecord(
         String fullName = composeName(parser.getPrefix(), localName);
         if (recordPathSegment.equals(localName) || recordPathSegment.equals(fullName)) {
             if (pathIndex < recordPath.length - 1) {
-                while (parser.hasNext() && limit != 0) {
-                    Token eventType = parser.next();
-                    if (eventType == Token.StartEntity) {
-                        findRecord(project, parser, recordPath, pathIndex + 1, rootColumnGroup, limit--,
-                                trimStrings, storeEmptyStrings, guessDataTypes);
-                    } else if (eventType == Token.EndEntity) {
-                        break;
-                    } else if (eventType == Token.Value) {
-                        // This is when the user picks a specific field to import, not a whole object or element.
-                        if (pathIndex == recordPath.length - 2) {
-                            String desiredFieldName = recordPath[pathIndex + 1];
-                            String currentFieldName = parser.getFieldName();
-                            if (desiredFieldName.equals(currentFieldName)) {
-                                processFieldAsRecord(project, parser, rootColumnGroup, trimStrings, storeEmptyStrings, guessDataTypes);
+                try {
+                    while (parser.hasNext() && limit != 0) {
+                        Token eventType = parser.next();
+                        if (eventType == Token.StartEntity) {
+                            findRecord(project, parser, recordPath, pathIndex + 1, rootColumnGroup, limit--,
+                                    trimStrings, storeEmptyStrings, guessDataTypes);
+                        } else if (eventType == Token.EndEntity) {
+                            break;
+                        } else if (eventType == Token.Value) {
+                            // This is when the user picks a specific field to import, not a whole object or element.
+                            if (pathIndex == recordPath.length - 2) {
+                                String desiredFieldName = recordPath[pathIndex + 1];
+                                String currentFieldName = parser.getFieldName();
+                                if (desiredFieldName.equals(currentFieldName)) {
+                                    processFieldAsRecord(project, parser, rootColumnGroup, trimStrings, storeEmptyStrings, guessDataTypes);
+                                }
                             }
                         }
                     }
+                } catch (TreeReaderException e) {
+                    logger.error("Exception from XML parse", e);
```

Comment:
That is not the best way to report errors to the user: in general, users should not have to look at the server logs to understand what is wrong. I would prefer to surface this message in the UI.

Reply:
Very true, am working on it